src/libxrpl/ledger/View.cpp

```diff
@@ -3277,6 +3277,42 @@ canTransfer(
     return tesSUCCESS;
 }
 
+[[nodiscard]] TER
+canTransfer(
+    ReadView const& view,
+    Issue const& issue,
+    AccountID const& from,
+    AccountID const& to)
+{
+    if (issue.native())
+        return tesSUCCESS;
+
+    auto const issuerId = issue.getIssuer();
+    auto const sleIssuer = view.read(keylet::account(issuerId));
+    if (sleIssuer == nullptr)
+        return tefINTERNAL;  // LCOV_EXCL_LINE
+    if (issuerId == from || issuerId == to)
+        return tesSUCCESS;
+
+    auto const isRippleDisabled = [&](AccountID account) -> bool {
+        // Line might not exist, but some transfers can create it. If this
+        // is the case, just check the default ripple on the issuer account.
+        auto const line = view.read(keylet::line(account, issue));
+        if (line)
+        {
+            bool const issuerHigh = issuerId > account;
+            return line->isFlag(issuerHigh ? lsfHighNoRipple : lsfLowNoRipple);
+        }
+        return sleIssuer->isFlag(lsfDefaultRipple) == false;
+    };
+
+    // Fail if rippling disabled on either trust line
+    if (isRippleDisabled(from) || isRippleDisabled(to))
+        return terNO_RIPPLE;
```

Comment:
This is not quite correct:

https://xrpl.org/docs/concepts/tokens/fungible-tokens/rippling#the-no-ripple-flag

> The No Ripple Flag
>
> Accounts that do not want to allow rippling can enable the No Ripple flag. If you enable the No Ripple flag on two of your trust lines, other accounts cannot use that pair of trust lines to ripple payments through your account. However, if you have any other trust lines that do not have the No Ripple flag enabled, *payments can still ripple through your account as long as one of the trust lines involved does not have No Ripple enabled.*
(emphasis added)

So this should be `&&`



```suggestion
    // Fail if rippling disabled on both trust lines
    if (isRippleDisabled(from) && isRippleDisabled(to))
        return terNO_RIPPLE;
```